Auto-generate documents for Script Resolver functions

Tkael opened the pull request:
Resolves #303
- Re-organizes custom script resolver functions into an interface and class structure
- Moves the script resolver itself into a sub-folder
- Generates Help.md and Functions.md automatically from classes during the post-build sequence

Tkael committed:
Update Cottle documentation web address

Tkael committed:
Refactor script resolver functions into classes and create a generator test to automatically update the documentation on functions.

Still TODO: Automatically update the local copy of Help.md when the app is built.

Tkael committed:
Auto-generate Help.md and Functions.md

Tkael committed:
Add annotations to custom functions

Tkael committed:
Flag the `VoiceDetail` class with the `[PublicAPI]` attribute.

Tkael commented on SpeechResponder/ScriptResolverService/ICustomFunction.cs:
```diff
@@ -0,0 +1,43 @@
+        Voice
+    }
+
+    public abstract class ResolverInstance<T1, T2>
+    {
+        protected ScriptResolver resolver { get; }
+        protected BuiltinStore store { get; }
+
+        protected ResolverInstance(T1 parameter1, T2 parameter2)
+        {
+            if (parameter1 is ScriptResolver resolverParameter)
+            {
+                resolver = resolverParameter;
+            }
+            if (parameter2 is BuiltinStore storeParameter)
+            {
+                store = storeParameter;
+            }
+        }
+    }
```
We're using the `ResolverInstance` abstract class to implement nesting / recursion for functions which require it.

richardbuckle commented on postBuildTests.bat:
```diff
@@ -17,7 +17,7 @@ IF %buildConfiguration%=="Release" (
   SET "testCaseFilter=^/TestCaseFilter:""TestCategory!=Speech"""
 ) ELSE (
   :: Run just our Credentials test
```
Need to update the comment, or simply remove it as it's not really adding much.

richardbuckle reviewed (changes requested):
I haven't read every single wrapped functionin detail.

richardbuckle commented:
Apart from the stale comment I'm happy.

richardbuckle commented on SpeechResponder/CustomFunctions/BlueprintDetails.cs:
```diff
@@ -0,0 +1,32 @@
+
+    {set blueprint to BlueprintDetails(""Dirty Drive Tuning"", 5)}
+    {len(blueprint.materials)} {if len(blueprint.materials) > 1: different materials are |else: material is} required to produce {blueprint.localizedName}.";
+        public NativeFunction function => new NativeFunction((values) =>
```
Sorry, a thought just struck me. Isn't this allocating a new instance of the NativeFunction every time the property is accessed? Don't we instead want to say:

```cs
    public NativeFunction function = new NativeFunction((values) =>
```
Or is this some form of [lazy initialization](https://docs.microsoft.com/en-us/dotnet/framework/performance/lazy-initialization)?

Tkael committed:
Fix stale comment

Tkael committed:
Add crowdin.yml to the solution

Tkael committed:
Move function description text to a resx file
(note: the file name is excluded from matching for CrowdIn translation - we'll keep the docs in English for the time being)
